packages/core/src/App/Containers/RealAccountSignup/account-wizard.jsx

```diff
@@ -418,15 +418,15 @@ const AccountWizard = observer(props => {
                     closeRealAccountSignup();
                     setShouldShowSameDOBPhoneModal(true);
                 } else if (modifiedProps.real_account_signup_target === 'maltainvest') {
+                    modifiedProps.onOpenDepositModal();
+                } else if (modifiedProps.real_account_signup_target === 'samoa') {
+                    modifiedProps.onOpenWelcomeModal(response.new_account_samoa.currency.toLowerCase());
+                } else {
+                    if (direct_deposit_flow) {
                         modifiedProps.onOpenDepositModal();
-                    } else if (modifiedProps.real_account_signup_target === 'samoa') {
```

Comment:
Do we need a sanity?

Reply:
Its after I merged master . Not related to my PR lemme check